snmalloc-sys/build.rs

```diff
@@ -1,335 +1,502 @@
+    match () {
+        _ if config.is_windows() => {
+            let common_flags = vec!["-mcx16", "-fno-exceptions", "-fno-rtti", "-pthread"];
+            for flag in common_flags {
+                config.builder.flag_if_supported(flag);
+            }
+
+            if let Some(msystem) = &config.msystem {
+                match msystem.as_str() {
+                    "CLANG64" | "CLANGARM64" => {
+                        let defines = vec![
+                            ("CMAKE_CXX_COMPILER", "clang++"),
+                            ("CMAKE_C_COMPILER", "clang"),
+                            ("CMAKE_CXX_FLAGS", "-fuse-ld=lld -stdlib=libc++ -mcx16 -Wno-error=unknown-pragmas -Qunused-arguments"),
+                            ("CMAKE_C_FLAGS", "-fuse-ld=lld -Wno-error=unknown-pragmas -Qunused-arguments"),
+                            ("CMAKE_EXE_LINKER_FLAGS", "-fuse-ld=lld -stdlib=libc++"),
+                            ("CMAKE_SHARED_LINKER_FLAGS", "-fuse-ld=lld -stdlib=libc++")
+                        ];
+                        apply_defines(&mut config.builder, &defines);
+                        if config.features.lto {
+                            config.builder.flag_if_supported("-flto=thin");
+                        }
+                    }
+                    "UCRT64" | "MINGW64" => {
+                        let defines = vec![
+                            ("CMAKE_CXX_FLAGS", "-fuse-ld=lld -Wno-error=unknown-pragmas"),
+                            ("CMAKE_SYSTEM_NAME", "Windows"),
+                            ("CMAKE_C_FLAGS", "-fuse-ld=lld -Wno-error=unknown-pragmas"),
+                            ("CMAKE_EXE_LINKER_FLAGS", "-fuse-ld=lld"),
+                            ("CMAKE_SHARED_LINKER_FLAGS", "-fuse-ld=lld")
+                        ];
+                        apply_defines(&mut config.builder, &defines);
+                    }
+                    _ => {}
+                }
+            }
+        }
+        _ if config.is_msvc() => {
+            let msvc_flags = vec![
+                "/nologo", "/W4", "/WX", "/wd4127", "/wd4324", "/wd4201",
+                "/Ob2", "/DNDEBUG", "/EHsc", "/Gd", "/TP", "/Gm-", "/GS",
+                "/fp:precise", "/Zc:wchar_t", "/Zc:forScope", "/Zc:inline"
+            ];
+            for flag in msvc_flags {
+                config.builder.flag_if_supported(flag);
+            }
+            
+            if config.features.lto {
+                config.builder
+                    .flag_if_supported("/GL")
+                    .define("CMAKE_INTERPROCEDURAL_OPTIMIZATION", "TRUE")
+                    .define("SNMALLOC_IPO", "ON");
+                println!("cargo:rustc-link-arg=/LTCG");
+            }
+            
+            config.builder
+                .define("CMAKE_CXX_FLAGS_RELEASE", "/O2 /Ob2 /DNDEBUG /EHsc")
+                .define("CMAKE_C_FLAGS_RELEASE", "/O2 /Ob2 /DNDEBUG /EHsc");
+        }
+        _ if config.is_linux() || config.is_unix() => {
```

Comment:
sorry to bother but isn't linux unix anyway ?

Reply:
In terms of cflags and cxxflags I didn't see much difference but interms of linking I ran into a slew of compatibility issues. Changes I made on Ubuntu didn't work on Macos, which was being classified as unix. so atomic, stdc++, etc...even between unix types there was varying compatibility or locations where produced libs will be located strangely, windows also had this issue.

Reply:
I think @devnexen  is asking for this test can it be simplified as
```suggestion
        _ if config.is_unix() => {
```

Reply:
Doh! you are correct, it's using CARGO_CFG_TARGET_FAMILY environment variable set by Rust's build system which does consider linux under unix family overall. Thanks for clarification!